File: Src/java/cql-to-elm/src/jvmTest/java/org/cqframework/cql/cql2elm/StringEscapeUtilsTest.java
package org.cqframework.cql.cql2elm;

import static org.cqframework.cql.cql2elm.StringEscapeUtils.escapeCql;
import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class StringEscapeUtilsTest {

    @Test
    void escape() {
        assertEquals("Hello \\'World\\'", escapeCql("Hello 'World'"));
        assertEquals("Hello \\\"World\\\"", escapeCql("Hello \"World\""));
        assertEquals("Hello \\`World\\`", escapeCql("Hello `World`"));
        assertEquals("Hello \\'World\\'2", escapeCql("Hello 'World'2"));
        assertEquals("Hello \\\"World\\\"2", escapeCql("Hello \"World\"2"));
        assertEquals("\\f\\n\\r\\t\\/\\\\", escapeCql("\f\n\r\t/\\"));
        assertEquals("\\u110f", escapeCql("ᄏ")); // unprintable character
        assertEquals(
                "This is an identifier with \\\"multiple\\\" embedded \\t escapes\u0020\\r\\nno really, \\r\\n\\f\\t\\/\\\\lots of them",
                escapeCql(
                        "This is an identifier with \"multiple\" embedded \t escapes\u0020\r\nno really, \r\n\f\t/\\lots of them"));
    }
}

File: Src/java/cql-to-elm/src/jvmTest/kotlin/org/cqframework/cql/cql2elm/StringEscapeUtilsTest.kt
package org.cqframework.cql.cql2elm

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

@Suppress("MaxLineLength")
internal class StringEscapeUtilsTest {
    @Test
    fun escape() {
        Assertions.assertEquals("Hello \\'World\\'", StringEscapeUtils.escapeCql("Hello 'World'"))
        Assertions.assertEquals(
            "Hello \\\"World\\\"",
            StringEscapeUtils.escapeCql("Hello \"World\"")
        )
        Assertions.assertEquals("Hello \\`World\\`", StringEscapeUtils.escapeCql("Hello `World`"))
        Assertions.assertEquals("Hello \\'World\\'2", StringEscapeUtils.escapeCql("Hello 'World'2"))
        Assertions.assertEquals(
            "Hello \\\"World\\\"2",
            StringEscapeUtils.escapeCql("Hello \"World\"2")
        )
        Assertions.assertEquals(
            "\\f\\n\\r\\t\\/\\\\",
            StringEscapeUtils.escapeCql("\u000c\n\r\t/\\")
        )
        Assertions.assertEquals(
            "\\u110f",
            StringEscapeUtils.escapeCql("ᄏ")
        ) // unprintable character
        Assertions.assertEquals(
            "This is an identifier with \\\"multiple\\\" embedded \\t escapes\u0020\\r\\nno really, \\r\\n\\f\\t\\/\\\\lots of them",
            StringEscapeUtils.escapeCql(
                "This is an identifier with \"multiple\" embedded \t escapes\u0020\r\nno really, \r\n\u000c\t/\\lots of them"
            )
        )
    }
}
